IndexError on extract_proper_type_queryset_values_list

Hello,

I updated to django-stubs 1.0.0 and mypy started crashing. Here is a minimal example that reproduces the issue:

```python
from decimal import Decimal

from django.db import models

class TransactionQuerySet(models.QuerySet):
    def amount_pending_repayments(self) -> Decimal:
        return self.values_list("total", flat=True)
```

It crashes with the following traceback:

```
INTERNAL ERROR -- Please try using mypy master on Github:
https://mypy.rtfd.io/en/latest/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 0.720
Traceback (most recent call last):
  File "mypy/checkexpr.py", line 3404, in accept
  File "mypy/nodes.py", line 1527, in accept
  File "mypy/checkexpr.py", line 234, in visit_call_expr__ExpressionVisitor_glue
  File "mypy/checkexpr.py", line 243, in visit_call_expr
  File "mypy/checkexpr.py", line 315, in visit_call_expr_inner
  File "mypy/checkexpr.py", line 670, in check_call_expr_with_callee_type
  File "mypy/checkexpr.py", line 728, in check_call
  File "mypy/checkexpr.py", line 835, in check_callable_call
  File "mypy/checkexpr.py", line 581, in apply_function_plugin
  File "/Users/canassa/.pyenv/versions/3.7.4/lib/python3.7/site-packages/mypy_django_plugin/transformers/querysets.py", line 98, in extract_proper_type_queryset_values_list
    if not isinstance(ctx.type.args[0], Instance):
IndexError: list index out of range
src/bli.py:8: : note: use --pdb to drop into pdb
```

And this is some `pdb` inspection:

```
src/bli.py:10: error: INTERNAL ERROR -- Please try using mypy master on Github:
https://mypy.rtfd.io/en/latest/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 0.720
Dropping into pdb
> /Users/canassa/.pyenv/versions/3.7.4/lib/python3.7/site-packages/mypy_django_plugin/transformers/querysets.py(148)extract_proper_type_queryset_values()
-> if not isinstance(ctx.type.args[0], Instance):
(Pdb) ctx
MethodContext(type=bli.TransactionQuerySet, arg_types=[[builtins.str], []], arg_kinds=[[0], []], callee_arg_names=['fields', 'expressions'], arg_names=[[None], []], default_return_type=django.db.models.query.QuerySet[Any, builtins.dict[builtins.str, Any]], args=[[<mypy.nodes.StrExpr object at 0x110069710>], []], context=<mypy.nodes.CallExpr object at 0x11007c150>, api=<mypy.checker.TypeChecker object at 0x112b8eb00>)
(Pdb) ctx.type
bli.TransactionQuerySet
(Pdb) ctx.type
bli.TransactionQuerySet
(Pdb) ctx.type.args
[]
(Pdb) c
```

--- a/mypy_django_plugin/transformers/meta.py
+++ b/mypy_django_plugin/transformers/meta.py
@@ -5,7 +5,7 @@ from mypy.types import Type as MypyType
 from mypy.types import TypeOfAny
 
 from mypy_django_plugin.django.context import DjangoContext
-from mypy_django_plugin.lib import fullnames, helpers
+from mypy_django_plugin.lib import helpers
 
 
 def _get_field_instance(ctx: MethodContext, field_fullname: str) -> MypyType:
@@ -20,21 +20,25 @@ def return_proper_field_type_from_get_field(ctx: MethodContext, django_context: 
     # Options instance
     assert isinstance(ctx.type, Instance)
 
+    # bail if list of generic params is empty
+    if len(ctx.type.args) == 0:
+        return ctx.default_return_type
+
     model_type = ctx.type.args[0]
     if not isinstance(model_type, Instance):
-        return _get_field_instance(ctx, fullnames.FIELD_FULLNAME)
+        return ctx.default_return_type
 
     model_cls = django_context.get_model_class_by_fullname(model_type.type.fullname())
     if model_cls is None:
-        return _get_field_instance(ctx, fullnames.FIELD_FULLNAME)
+        return ctx.default_return_type
 
     field_name_expr = helpers.get_call_argument_by_name(ctx, 'field_name')
     if field_name_expr is None:
-        return _get_field_instance(ctx, fullnames.FIELD_FULLNAME)
+        return ctx.default_return_type
 
     field_name = helpers.resolve_string_attribute_value(field_name_expr, ctx, django_context)
     if field_name is None:
-        return _get_field_instance(ctx, fullnames.FIELD_FULLNAME)
+        return ctx.default_return_type
 
     try:
         field = model_cls._meta.get_field(field_name)

--- a/mypy_django_plugin/transformers/querysets.py
+++ b/mypy_django_plugin/transformers/querysets.py
@@ -3,6 +3,7 @@ from typing import List, Optional, Sequence, Type, Union, cast
 
 from django.core.exceptions import FieldError
 from django.db.models.base import Model
+from django.db.models.fields.related import RelatedField
 from mypy.newsemanal.typeanal import TypeAnalyser
 from mypy.nodes import Expression, NameExpr, TypeInfo
 from mypy.plugin import AnalyzeTypeContext, FunctionContext, MethodContext
@@ -10,9 +11,17 @@ from mypy.types import AnyType, Instance
 from mypy.types import Type as MypyType
 from mypy.types import TypeOfAny
 
-from django.db.models.fields.related import RelatedField
 from mypy_django_plugin.django.context import DjangoContext
 from mypy_django_plugin.lib import fullnames, helpers
+
+
+def _extract_model_type_from_queryset(queryset_type: Instance) -> Optional[Instance]:
+    for base_type in [queryset_type, *queryset_type.type.bases]:
+        if (len(base_type.args)
+                and isinstance(base_type.args[0], Instance)
+                and base_type.args[0].type.has_base(fullnames.MODEL_CLASS_FULLNAME)):
+            return base_type.args[0]
+    return None
 
 
 def determine_proper_manager_type(ctx: FunctionContext) -> MypyType:
@@ -98,11 +107,10 @@ def extract_proper_type_queryset_values_list(ctx: MethodContext, django_context:
     assert isinstance(ctx.type, Instance)
     assert isinstance(ctx.default_return_type, Instance)
 
-    # bail if queryset of Any or other non-instances
-    if not isinstance(ctx.type.args[0], Instance):
+    model_type = _extract_model_type_from_queryset(ctx.type)
+    if model_type is None:
         return AnyType(TypeOfAny.from_omitted_generics)
 
-    model_type = ctx.type.args[0]
     model_cls = django_context.get_model_class_by_fullname(model_type.type.fullname())
     if model_cls is None:
         return ctx.default_return_type
@@ -148,11 +156,10 @@ def extract_proper_type_queryset_values(ctx: MethodContext, django_context: Djan
     assert isinstance(ctx.type, Instance)
     assert isinstance(ctx.default_return_type, Instance)
 
-    # if queryset of non-instance type
-    if not isinstance(ctx.type.args[0], Instance):
+    model_type = _extract_model_type_from_queryset(ctx.type)
+    if model_type is None:
         return AnyType(TypeOfAny.from_omitted_generics)
 
-    model_type = ctx.type.args[0]
     model_cls = django_context.get_model_class_by_fullname(model_type.type.fullname())
     if model_cls is None:
         return ctx.default_return_type

--- a/setup.py
+++ b/setup.py
@@ -28,7 +28,7 @@ dependencies = [
 
 setup(
     name="django-stubs",
-    version="1.0.1",
+    version="1.0.2",
     description='Mypy stubs for Django',
     long_description=readme,
     long_description_content_type='text/markdown',

--- a/test-data/typecheck/managers/querysets/test_values_list.yml
+++ b/test-data/typecheck/managers/querysets/test_values_list.yml
@@ -205,3 +205,21 @@
                     pass
                 class Blog(models.Model):
                     publisher = models.ForeignKey(to=Publisher, on_delete=models.CASCADE)
+
+-   case: subclass_of_queryset_has_proper_typings_on_methods
+    main: |
+        from myapp.models import TransactionQuerySet
+        reveal_type(TransactionQuerySet())  # N: Revealed type is 'myapp.models.TransactionQuerySet'
+        reveal_type(TransactionQuerySet().values())  # N: Revealed type is 'django.db.models.query.QuerySet[myapp.models.Transaction, TypedDict({'id': builtins.int, 'total': builtins.int})]'
+        reveal_type(TransactionQuerySet().values_list())  # N: Revealed type is 'django.db.models.query.QuerySet[myapp.models.Transaction, Tuple[builtins.int, builtins.int]]'
+    installed_apps:
+        - myapp
+    files:
+        -   path: myapp/__init__.py
+        -   path: myapp/models.py
+            content: |
+                from django.db import models
+                class TransactionQuerySet(models.QuerySet['Transaction']):
+                    pass
+                class Transaction(models.Model):
+                    total = models.IntegerField()

